src/lib/market-service/market-service-manager.ts

```diff
@@ -93,7 +94,30 @@ export class MarketServiceManager {
         // Swallow error, not every asset will be with every provider.
       }
     }
-    if (!result) return null
+
+    // If we don't find any results, then we look for related assets
+    if (!result) {
+      const relatedAssetIds = await _getRelatedAssetIds(assetId)
```

Comment:
should this call RTK query to utilize the cache a layer higher? this data won't change and will make network requests

Reply:
This has been extracted because of circular deps, calling the zerion API from the market service will produce big rugeroo - can take another look in the mo' at fixing circular deps if that's realistically possible

Reply:
Actually `marketApi.endpoints.findByAssetId` calls are cached themselves so we're good - if they weren't, we'd had bigger problems and spam the market data providers like crazy currently

Reply:
2e3899b994d27ab1975755e41789fb99d7283de3